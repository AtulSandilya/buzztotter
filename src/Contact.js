import { Image, Text, View } from 'react-native';
import React, { Component, PropTypes } from 'react';

export default class Contact extends Component {
  static propTypes = {
    name: React.PropTypes.string,
    birthday: React.PropTypes.string,
    imagePath: React.PropTypes.string,
  }

  render() {
    return(
      <View style={{flex: 1}}>
        <Image
          source={require('../img/icons/bev-contact.png')}
        />
        <Text>{this.props.name}</Text>
        <Text>{this.props.birthday}</Text>
        <Text>Buy Beer</Text>
      </View>
    );
  }
}
