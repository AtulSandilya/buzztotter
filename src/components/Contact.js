import { Image, StyleSheet, Text, TouchableHighlight, View } from 'react-native';
import React, { Component, PropTypes } from 'react';

import BevButton from './BevButton';

const styles = StyleSheet.create({
  parentContainer: {
    flex: 1,
    flexDirection: 'row',
  },
  infoContainer: {
    flex: 2,
    flexDirection: 'row',
    alignSelf: 'center',
    alignItems: 'center',
    paddingLeft: 10,
  },
  infoTextContainer: {
    flex: 1,
    flexDirection: 'column',
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'row',
    alignSelf: 'center',
    alignItems: 'center',
  },
})

const Contact = ({name, birthday, openPurchaseModal, closePurchaseModal}) => {
  const fullName = name.first + " " + name.last;
  return (
    <View style={styles.parentContainer}>
      <View style={styles.infoContainer}>
        <Image
          source={require('../../img/icons/bev-contact.png')}
        />
        <View style={styles.infoTextContainer}>
          <Text style={{paddingLeft: 15, paddingBottom: 5}}>{fullName}</Text>
          <Text style={{paddingLeft: 15}}>{birthday}</Text>
        </View>
      </View>
      <View style={styles.buttonContainer}>
        <BevButton
          buttonText={"Send " + name.first + " a Beer!"}
          bevButtonPressed={() => openPurchaseModal({fullName: fullName, firstName: name.first})}
        />
      </View>
    </View>
  )
}

export default Contact;
