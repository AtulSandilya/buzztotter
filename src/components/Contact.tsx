import * as React from "react";
import { Component } from 'react';
import { Image, StyleSheet, Text, TouchableHighlight, View, ViewStyle } from 'react-native';

import BevButton from './BevButton';

interface Styles {
  parentContainer: ViewStyle;
  infoContainer: ViewStyle;
  infoTextContainer: ViewStyle;
  buttonContainer: ViewStyle;
}

const styles = StyleSheet.create<Styles>({
  parentContainer: {
    flex: 1,
    flexDirection: 'row',
  },
  infoContainer: {
    flex: -1,
    flexDirection: 'row',
    alignItems: 'center',
    alignSelf: 'center',
    paddingLeft: 10,
  },
  infoTextContainer: {
    flex: -1,
    flexDirection: 'column',
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'row',
    alignSelf: 'center',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
})

interface Name {
  first: string;
  last: string;
}

export interface ContactProps {
  name?: Name;
  birthday?: string;
  imagePath?: string;
  facebookId?: string;
  openPurchaseRoute?(Object): void;
  closePurchaseRoute?(): void;
}

const Contact: React.StatelessComponent<ContactProps> = ({name, birthday, imagePath, facebookId, openPurchaseRoute, closePurchaseRoute}) => {
  const fullName = name.first + " " + name.last;
  return (
    <View style={styles.parentContainer}>
      <View style={styles.infoContainer}>
        <Image
          source={{uri: imagePath}}
          style={{height: 50, width: 50}}
        />
        <View style={styles.infoTextContainer}>
          <Text style={{paddingLeft: 15, paddingBottom: 5}}>{fullName}</Text>
          <Text style={{paddingLeft: 15}}>{birthday}</Text>
        </View>
      </View>
      <View style={styles.buttonContainer}>
        <BevButton
          text={"Send Bevegram"}
          shortText="Send Bevegram"
          label="Send Bevegram Button"
          onPress={() => openPurchaseRoute({
            fullName: fullName,
            firstName: name.first,
            imageUri: imagePath,
            facebookId: facebookId,
          })}
          rightIcon={true}
        />
      </View>
    </View>
  )
}

export default Contact;
