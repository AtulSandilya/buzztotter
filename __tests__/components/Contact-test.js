import 'react-native';
import React from 'react';
import Contact from '../../build/components/Contact';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Contact
      name={{first: "Andrew", last: "Simms"}}
      birthday="12/12"
      imagePath="test.jpg"
      openPurchaseModal={() => {}}
      closePurchaseModal={() => {}}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

// it('fails', () => {
//   const tree = renderer.create(
//     <Contact
//     />
//   ).toJSON();
//   expect(tree).toThrowErrorMatchingSnapshot();
// });
