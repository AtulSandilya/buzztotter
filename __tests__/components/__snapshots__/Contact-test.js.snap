// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "padding": 4.888888888888888,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "elevation": 3,
            "margin": 4.888888888888888,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.18,
            "shadowRadius": 2,
          },
          Object {},
          Object {
            "borderRadius": 22.734326675206372,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "test.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 22.734326675206372,
            "height": 45.468653350412744,
            "width": 45.468653350412744,
            "zIndex": 1000,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": -1,
          "flexDirection": "column",
          "paddingLeft": 11,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#333333",
            "fontFamily": undefined,
            "fontSize": 15,
            "fontWeight": "normal",
            "letterSpacing": undefined,
            "lineHeight": 22.5,
            "paddingBottom": 3.259259259259259,
            "textAlignVertical": "center",
          }
        }
        textStyle={
          Object {
            "paddingBottom": 3.259259259259259,
          }
        }
      >
        Andrew Simms
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": -1,
              "flexDirection": "row",
              "margin": 0,
              "padding": 0,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "paddingRight": 7.333333333333333,
                "width": undefined,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#8D8D9D",
                  "fontSize": 15,
                  "lineHeight": 15,
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          accessible={true}
          allCaps={true}
          allowFontScaling={true}
          color="#51515E"
          disabled={false}
          ellipsizeMode="tail"
          fontWeight="bold"
          isCondensed={true}
          size="small"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#51515E",
              "fontFamily": "Roboto",
              "fontSize": 10,
              "fontWeight": "bold",
              "letterSpacing": 0.5,
              "lineHeight": 10,
              "margin": 0,
              "paddingBottom": 0,
              "paddingTop": 2,
              "textAlignVertical": "center",
            }
          }
          textStyle={
            Object {
              "margin": 0,
              "paddingBottom": 0,
              "paddingTop": 2,
            }
          }
          underline={false}
        >
          12/12
        </Text>
        <View />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": -1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "elevation": 3,
            "margin": 4.888888888888888,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.18,
            "shadowRadius": 2,
          },
          Object {},
          Object {
            "borderRadius": 3,
          },
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "backgroundColor": "#8B5E3C",
              "borderColor": "#8B5E3C",
              "borderRadius": 3,
              "borderWidth": 0.5,
              "flex": -1,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": -1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "paddingHorizontal": 11,
                  "paddingVertical": 7.333333333333333,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#ffffff",
                    "fontSize": 15,
                    "lineHeight": 15,
                    "textShadowColor": "rgba(0, 0, 0, 0.25)",
                    "textShadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "textShadowRadius": 2,
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            accessible={true}
            allCaps={false}
            allowFontScaling={true}
            color="#ffffff"
            disabled={false}
            ellipsizeMode="tail"
            fontWeight="bold"
            isCondensed={true}
            showTextShadow={true}
            size="normal"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 15,
                "fontWeight": "bold",
                "letterSpacing": undefined,
                "lineHeight": 22.5,
                "paddingLeft": 0,
                "paddingRight": 11,
                "paddingVertical": 7.333333333333333,
                "textAlignVertical": "center",
                "textShadowColor": "rgba(0, 0, 0, 0.25)",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "textShadowRadius": 2,
                "top": 0,
              }
            }
            textStyle={
              Object {
                "paddingLeft": 0,
                "paddingRight": 11,
                "paddingVertical": 7.333333333333333,
                "top": 0,
              }
            }
          >
            Send Bevegram
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
